src/main/java/com/unb/tracker/web/CourseController.java

```diff
@@ -163,7 +163,16 @@ public String courseSave(@ModelAttribute Course course, ModelMap map, Principal
             redir.addFlashAttribute("courseRows", course.getRows());
             redir.addFlashAttribute("courseCols", course.getCols());
 
-            return "redirect:/" + user.getUsername();
+            if(course.getId() == null) {
+                return "redirect:/" + user.getUsername();
+            } else {
+                String url = user.getUsername() + "/" + course.getName();
```

Comment:
I think here if the course.getName() returns null we need to get the actual course from the DB and redir there instead, because right now the redirect doesn't work if you leave name empty.